### ISSUE TITLE
Sección Archivos en el enunciado

### ISSUE THREAD

**Issue by JosefaATP** [IIC2233/Syllabus#606]
### Prerrequisitos
(Marcar colocando una `X` entre los corchetes los ítems que ya hiciste, así: `"[X]"`)

* [ X ] Leí las reglas del foro (https://github.com/IIC2233/Syllabus/wiki/3.-Reglas-del-foro-(issues)).
* [ X ] Revisé el compilado de dudas de la tarea y no encontré una *issue* similar a la mía (https://github.com/IIC2233/Syllabus/issues/421).
* [ X ] Mi duda no se trata sobre una librería o *built-in*, ya que eso se pregunta en la *issue* creada para ello ([click aquí para ir a esa _issue_](/IIC2233/Syllabus/issues/420)).
* [ X ] De ser necesario, colocaré código simple que permita explicar mi problema o duda, ya que compartir código de esta evaluación puede ser considerado una falta a la ética.

### Duda

Hola!!

No entiendo en general la sección archivos del enunciado, en términos generales entiendo para que son:
1- "deben ser almacenados de manera correcta y consistente. Esto quiere decir de que inicialmente solo deben existir en algún directorio , cuyo nombre queda a criterio del lector y debe ser especificado en el archivo parametros.py, que se ubique dentro del directorio del Servidor Principal."  Esto quiere decir que la carpeta que uno se descarga debe ir en la carpeta del servidor?

2- "Luego, a medida que algún cliente descargue exitosamente los archivos, estos deben quedar almacenados dentro de algún directorio, cuyo nombre queda a criterio del lector y debe ser especificado en el archivo parametros.py" Esto quiere decir que se deben guardar una copia en una carpeta del cliente, debido a que "estamos en compus distintos"? 

3- ¿Que sucede en el caso de varios clientes? donde se deberian guardar? porque no es que haga una carpeta por cliente o si?

4- Y estoy casi segura que algo leí del nombre de los archivos que uno se descargo. Que era algo como que se podía cambiar el nombre de los archivos individuales? Si es así, para que sería? Que beneficio tiene?

Muchas dudas ejejej 
Gracias!!🤓

**Comment by 3rdPix** [comment]
Hola @JosefaATP !

1. Correcto, va a la carpeta servidor, ahí comienzan todos los archivos
2. Correcto! Son computadores distintos, por lo que no pueden compartir archivos como si estuvieran en la misma máquina
3. En el caso de varios clientes, lo que puedes hacer es ejecutar un cliente con un directorio, y luego un segundo cliente cambiando el parámetro para que trabajen en directorios distintos, también son computadores distintos en teoría
4. No entendí bien la pregunta... a qué te refieres?

**Comment by JosefaATP** [comment]
Osea seria algo asi: ??
T4
|- cliente
      |- backend
      |- ....
|- servidor
      | - archivos


4. Me referia a que por ejemplo al descargar la carpeta de archivos, uno de ellos se llama: pepa_power.jepg. Que a esta por ejemplo le cambio el nombre a 01.jpeg. Ahí si?

**Comment by 3rdPix** [comment]
Tu árbol de directorio parece estar correcto!

4. En principio no debería suponer ningún problema, porque los archivos debes agregarlos al `.gitignore` (**por favor hazlo**). Sin embargo, ten en consideración que tu programa no puede depender de los nombres de archivos, este debería ser capaz de leer cualquier archivo que tengas en el directorio de `data/`, y el cliente debería poder descargar cualquiera de ellos sin importar el nombre. No puedes _hardcodear_ los nombres de los archivos. Aún así, reitero, puedes cambiar el nombre si quieres facilitar tus pruebas en el programa, así no tienes que escribir nombres tan largos cada vez que quieras probar algo.

Me cuentas si te quedó más claro :D

**Comment by JosefaATP** [comment]
Ah, yayay perfecto!
Ajajajjaja sisi, ya estan en el .gitignore

Si, me quedo más claro.

Gracias!🤓

**Comment by Delfin13120** [comment]
> Hola @JosefaATP !
> 
> 1. Correcto, va a la carpeta servidor, ahí comienzan todos los archivos
> 2. Correcto! Son computadores distintos, por lo que no pueden compartir archivos como si estuvieran en la misma máquina
> 3. En el caso de varios clientes, lo que puedes hacer es ejecutar un cliente con un directorio, y luego un segundo cliente cambiando el parámetro para que trabajen en directorios distintos, también son computadores distintos en teoría
> 4. No entendí bien la pregunta... a qué te refieres?

Holaa! Tengo una duda con respecto a esta respuesta que se dio:

En el punto 3, cuando se refiere a que hay que ejecutar un cliente cambiando el parámetro del dirctorio en el que se guardan los archivos, significa que a medida que se vayan conectando clientes, dentro de la carpeta T4/Cliente se tienen que generar directorios distintos para cada cliente y así cada uno puede manejar sus archivos de manera independiente? Porque sino no entiendo como se podría hacer que cada cliente tenga su propia carpeta de archivos descargados.

Saludos!!

**Comment by 3rdPix** [comment]
@Delfin13120

Exacto! Tal como sugieres, es necesario que cada cliente tenga su propia carpeta de archivos: en principio cada cliente es un computador diferente, por lo que no deberían tener acceso a los mismo archivos como si compartieran el mismo computador.

Hay varias formas de lograr esto:
- copiando tu cliente a un directorio distinto y ejecutarlo desde ahí (suponiendo que tu forma de manejar el _path_ al directorio de archivos esté construido relativo al programa, entonces trabajaría sobre ese directorio donde lo copiaste sin colisionar con los otros clientes)
- cambiando el parámetro con el _path_ al directorio de archivos: si ejecutas un cliente python inmediatamente carga a la memoria esos parámetros, por lo que si cambias un valor y ejecutas un nuevo cliente, aunque lo ejecutes desde el mismo directorio, utilizarán parámetros distintos

Me cuentas si tienes mas dudas!
